File: app/src/main/java/com/example/android/ecommerce/model/Category.java
package com.example.android.ecommerce.model;


import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

public class Category {

    @SerializedName("id")
    @Expose
    private long id;
    @SerializedName("name")
    @Expose
    private String name;
    @SerializedName("products")
    @Expose
    private List<Product> products = null;
    @SerializedName("child_categories")
    @Expose
    private List<Object> childCategories = null;

    /**
     * No args constructor for use in serialization
     */
    public Category() {
    }

    /**
     * @param name
     * @param childCategories
     * @param id
     * @param products
     */
    public Category(long id, String name, List<Product> products, List<Object> childCategories) {
        super();
        this.id = id;
        this.name = name;
        this.products = products;
        this.childCategories = childCategories;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public List<Product> getProducts() {
        return products;
    }

    public void setProducts(List<Product> products) {
        this.products = products;
    }

    public List<Object> getChildCategories() {
        return childCategories;
    }

    public void setChildCategories(List<Object> childCategories) {
        this.childCategories = childCategories;
    }
}

File: app/src/main/java/com/example/android/ecommerce/model/Category.kt
package com.example.android.ecommerce.model

import android.annotation.SuppressLint
import android.os.Parcelable
import androidx.room.Entity
import androidx.room.PrimaryKey
import androidx.room.TypeConverters
import com.example.android.ecommerce.typeconverters.ChildCategoryListTypeConverter
import com.example.android.ecommerce.typeconverters.ProductListTypeConverter
import com.google.gson.annotations.SerializedName
import kotlinx.android.parcel.Parcelize

@Entity
@SuppressLint("ParcelCreator")
@Parcelize
data class Category(
    @TypeConverters(ChildCategoryListTypeConverter::class)
    @SerializedName("child_categories")
    var childCategories: MutableList<Long?>? = null,
    @SerializedName("id")
    @PrimaryKey (autoGenerate = true)
    var id: Long = 0, // 13
    @SerializedName("name")
    var name: String? = null, // Laptops
    @TypeConverters(ProductListTypeConverter::class)
    @SerializedName("products")
    var products: MutableList<Product?>? = null
) : Parcelable